lib/improver/nbhood/square_kernel.py

```diff
@@ -315,25 +312,13 @@ def remove_halo_from_cube(self, cube, width_x, width_y):
             cube, trimmed_data, trimmed_x_coord, trimmed_y_coord)
         return trimmed_cube
 
-    def mean_over_neighbourhood(self, cube, cells_x, cells_y, nan_masks):
+    @staticmethod
+    def calculate_neighbourhood(summed_cube,
+                                ymax_xmax_disp, ymin_xmax_disp,
+                                ymin_xmin_disp, ymax_xmin_disp,
+                                n_rows, n_columns):
         """
-        Method to calculate the average value in a square neighbourhood using
-        the 4-point algorithm to find the total sum over the neighbourhood.
-
-        The output from the cumulate_array method can be used to
-        calculate the sum over a neighbourhood of size
-        (2*cells_x+1)*(2*cells_y+1). This sum is then divided by the area of
-        the neighbourhood to calculate the mean value in the neighbourhood.
-
-        For all points, a fast vectorised approach is taken:
-
-        1. The displacements between the four points used to calculate the
-           neighbourhood total sum and the central grid point are calculated.
-        2. Four copies of the cumulate array output are flattened and rolled
-           by these displacements to align the four terms used in the
-           neighbourhood total sum calculation.
-        3. The neighbourhood total at all points can then be calculated
-           simultaneously in a single vector sum.
+        Fast vectorised approach to calculating neighbourhood totals.
 
         Displacements are calculated as follows for the following input array,
         where the accumulation has occurred from left to right and top to
```

Comment:
I'm not sure the orientation of this array in this docstring matches that in the unit test for this function. It doesn't help explain the result in the unit test for this function.

Reply:
I agree that the orientation doesn't match the way it works in reality but I haven't changed this docstring. It has always been the wrong way up. I'll reorder it if you think it makes it clearer.